### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Abelian): AB axioms carry over to functor categories

### PR DESCRIPTION
**Pull request by dagurtomas**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Abelian/GrothendieckAxioms/FunctorCategory.lean [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+namespace CategoryTheory
+
+open CategoryTheory Limits Opposite
+
+section
+
+variable {A C J : Type*} [Category A] [Category C] [Category J]
+    [HasColimitsOfShape J A] [HasExactColimitsOfShape J A] [HasFiniteLimits A]
+
+instance : HasExactColimitsOfShape J (C ⥤ A) where
```
Comment (TwoFX) [inline Mathlib/CategoryTheory/Abelian/GrothendieckAxioms/FunctorCategory.lean:24]: I wonder if this follows immediately from https://github.com/leanprover-community/mathlib4/blob/415efa61fc0b3ee799f90fe287954d6016bd1bac/Mathlib/CategoryTheory/Limits/Preserves/FunctorCategory.lean#L225

Comment (dagurtomas) [reply Mathlib/CategoryTheory/Abelian/GrothendieckAxioms/FunctorCategory.lean:24]: Indeed, it does! Thanks

File: Mathlib/CategoryTheory/Abelian/GrothendieckAxioms/Basic.lean [hunk 2]
```diff
@@ -79,6 +79,48 @@ class HasExactLimitsOfShape (J : Type u') [Category.{v'} J] (C : Type u) [Catego
 attribute [instance] HasExactColimitsOfShape.preservesFiniteLimits
   HasExactLimitsOfShape.preservesFiniteColimits
 
+variable {C} in
+/--
+Pull back a `HasExactColimitsOfShape J` along a functor which preserves and reflects finite limits
+and preserves colimits of shape `J`
+-/
+lemma hasExactColimitsOfShape_transport {D : Type*} (J : Type*) [Category J] [Category D]
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Abelian/GrothendieckAxioms/Basic.lean:87]: My only question is whether `_transport` is the optimal suffix. How about
```suggestion
lemma HasExactColimitsOfShape.domain_of_functor {D : Type*} (J : Type*) [Category J] [Category D]
```
or something like that?

`transport` suggest an equivalence to me, and your statement asks for less...

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**